lib/kubeclient/common.rb

```diff
@@ -68,6 +68,7 @@ def handle_exception
         {}
       end
       err_message = json_error_msg['message'] || e.message
+      raise NameError.new if json_error_msg && json_error_msg['reason'] == 'NotFound'
```

Comment:
Right :(
The error message is different. I'm wondering If we could use that?
For non existing types we get:
```"message": "the server could not find the requested resource",```
For non existing entities:
```"message": "nodes \"127.0.0.12\" not found",```